### DISCUSSION TITLE
R - Gagal melakukan proses summarise kepada beberapa kolom sekaligus

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#17] [Q&A] [answered]
Diketahui, data frame dengan struktur berikut:
str(KC_clean)
tibble [30 x 12] (S3: tbl_df/tbl/data.frame)
 $ pos      : num [1:30] 19 19 13 16 19 19 19 13 19 18 ...
 $ team     : chr [1:30] "Kopi_Contrapuntal" "Kopi_Contrapuntal" "Kopi_Contrapuntal" "Kopi_Contrapuntal" ...
 $ pld      : num [1:30] 36 36 36 36 36 36 36 36 36 36 ...
 $ won      : num [1:30] 5 6 12 11 6 5 5 13 8 7 ...
 $ drn      : num [1:30] 10 8 7 7 9 9 5 4 8 7 ...
 $ lst      : num [1:30] 21 22 17 18 21 22 26 19 20 22 ...
 $ for      : num [1:30] 39 55 59 80 63 58 46 59 60 61 ...
 $ ag       : num [1:30] 77 110 77 91 93 102 98 78 87 92 ...
 $ gd       : num [1:30] -38 -55 -18 -11 -30 -44 -52 -19 -27 -31 ...
 $ pts      : num [1:30] 25 26 43 40 27 24 20 43 32 28 ...
 $ side_role: chr [1:30] "WM AT" "WM AT" "WM AT" "WM AT" ...
 $ sim      : num [1:30] 1 2 3 4 5 6 7 8 9 10 ...

![kc clean](https://user-images.githubusercontent.com/95233522/158998356-bac3aed0-6534-4b5f-b683-3d5437be2626.JPG)

Nah saya mau buat summarise berdasarkan side role jadi rata-rata dan persentase berikut, namun gagal. Kenapa ya, seinget saya bisa dibuat beberapa item saat bikin summarise. 

KC_by_role_avg <- KC_clean %>%
  group_by(side_role) %>%
  summarise(
  avg_pos = mean(pos),
  avg_goal_for = mean(for),
  avg_goal_against = mean(ag),
  percent_won = sum(won)/sum(pld), 
  percent_draw = sum(drn)/sum(pld), 
  percent_lost = sum(lst)/sum(pld),
  )

Error messagenya: 

Error: unexpected ')' in:
"  avg_pos = mean(pos),
  avg_goal_for = mean(for)"
>   avg_goal_against = mean(ag),
Error: unexpected ',' in "  avg_goal_against = mean(ag),"
>   percent_won = sum(won)/sum(pld), 
Error: unexpected ',' in "  percent_won = sum(won)/sum(pld),"
>   percent_draw = sum(drn)/sum(pld), 
Error: unexpected ',' in "  percent_draw = sum(drn)/sum(pld),"
>   percent_lost = sum(lst)/sum(pld),
Error: unexpected ',' in "  percent_lost = sum(lst)/sum(pld),"
>   )
Error: unexpected ')' in "  )"

----

Karena gagal, akhirnya saya buat terpisah

KC_sum_by_role_avg <- KC_clean %>%
  group_by(side_role) %>%
  summarise_at(c("pos", "for", "ag", "gd"), mean, na.rm = TRUE)

#kenapa bisa?
KC_sum_by_role_percent <- KC_clean %>%
  group_by(side_role) %>%
  summarise(
  percent_won = round(100*(sum(won)/sum(pld)),digits=2), 
  percent_draw = round(100*(sum(drn)/sum(pld)),digits=2), 
  percent_lost = round(100*(sum(lst)/sum(pld)),digits=2),
)

apa ada cara yang lebih efisien biar sekaligus dan jadi 1 tabel untuk summarisenya?

**Comment by Davidlimbong** [comment, marked as answer]
Hi Hertzspr, error diatas disebabkan 2 hal 
1. Penggunaan for sebagai nama kolom
2. Kelebihan penggunaan koma pada `percent_lost = sum(lst)/sum(pld),`

dibawah ini solusi code dari kedua error diatas


```
KC_by_role_avg <- KC_clean %>%
group_by(side_role) %>%
  summarise(
    avg_pos = mean(pos),
    avg_goal_for = mean(`for`),
    avg_goal_against = mean(ag),
    percent_won = sum(won)/sum(pld),
    percent_draw = sum(drn)/sum(pld),
    percent_lost = sum(lst)/sum(pld)
)
```

Anda bisa mengapit nama kolom **for** dengan backtick ` serta menghapus tanda koma pada baris code terakhir.

**Reply by Hertzspr** [reply]
Terima kasih

**Reply by Hertzspr** [reply]
for itu nggak bisa karena dia keyword ya.. pantes warnanya beda di RStudio, saya khilaf
